packages/cursorless-engine/src/api/CursorlessEngineApi.ts

```diff
@@ -34,15 +34,28 @@ export interface CommandApi {
    * Runs a command.  This is the core of the Cursorless engine.
    * @param command The command to run
    */
-  runCommand(command: Command): Promise<unknown>;
+  runCommand(command: Command): Promise<CommandResponse | unknown>;
 
   /**
    * Designed to run commands that come directly from the user.  Ensures that
    * the command args are of the correct shape.
    */
-  runCommandSafe(...args: unknown[]): Promise<unknown>;
+  runCommandSafe(...args: unknown[]): Promise<CommandResponse | unknown>;
 }
 
+export type CommandResponse = { returnValue: unknown } | { fallback: Fallback };
+
+export type Fallback =
+  | { action: string; scope: string | null }
```

Comment:
maybe make the `string` into a stronger type

Reply:
done